Makefile

```diff
@@ -30,3 +30,9 @@ bash/fmt:
 
 bash/fmt/check:
 	shfmt -d $(PWD)/rootfs
+
+apk-update:
```

Comment:
Not crazy about the external (somewhat circular) dependency introduced here. Would really prefer the `ADD` solution if it works since that's a one-liner.

Reply:
Also, the `ADD` workaround if it works is fully self-contained and doesn't have "hooks" outside of the `Dockerfile`

Reply:
Using `sed` to edit the repositories file made it look to me like you strongly wanted to avoid explicitly listing URLs, and ADD will only work if you are going to fully specify the URL for each repo index file. If you want to do that, then yes, that is a good solution because it is self contained, although it will very much slow me down because on every build it will have to fetch all the indexes just to verify that they haven't changed. 

The `make` approach lets you explicitly determine when to update packages, and effectively pins the packages in between updates to the md5 fie.